File: dongyimai-interface/src/main/java/org/example/liuhengfei/service/TbGoodsService.java
package org.example.liuhengfei.service;

import org.example.liuhengfei.dto.PageResult;
import org.example.liuhengfei.pojo.TbGoods;

import java.util.List;

/**
 * 服务层接口
 *
 * @author Administrator
 */
public interface TbGoodsService {

    /**
     * 返回全部列表
     *
     * @return
     */
    List<TbGoods> findAll();

    /**
     * 返回分页列表
     *
     * @return
     */
    PageResult findPage(int pageNum, int pageSize, TbGoods tbGoods);

    /**
     * 根据ID获取实体
     *
     * @param id
     * @return
     */
    TbGoods findOne(Long id);

    TbGoods findOne(Long id, String sellerId);

    /**
     * 增加、修改
     */
    void save(TbGoods tbGoods, String sellerId);

    /**
     * 批量删除
     *
     * @param ids
     */
    void delete(Long[] ids);

    void updateStatus(Long[] ids, String status);
}

File: dongyimai-manager-web/src/main/webapp/js/service/TbGoodsService.js
//服务层
app.service("TbGoodsService", function ($http) {

    //读取列表数据绑定到表单中
    this.findAll = function () {
        return $http.post("/TbGoodsController/findAll");
    }
    //分页
    this.findPage = function (pageNum, pageSize, tbGoods) {
        return $http.post("/TbGoodsController/findPage?pageNum=" + pageNum + "&pageSize=" + pageSize, tbGoods);
    }
    //查询实体
    this.findOne = function (id) {
        return $http.post("/TbGoodsController/findOne?id=" + id);
    }
    //批量修改审核状态
    this.updateStatus = function (ids, status) {
        return $http.post("/TbGoodsController/updateStatus?ids=" + ids + "&status=" + status);
    }
    //删除
    this.delete = function (ids) {
        return $http.post("/TbGoodsController/delete?ids=" + ids);
    }
});